plugin/face-mask-filter.cpp

```diff
@@ -1167,19 +1196,36 @@ void Plugin::FaceMaskFilter::Instance::video_render(gs_effect_t *effect) {
 	
 	// demo mode render stuff
 		// generate previews?
-	if ( recordTriggered || (demoModeGenPreviews && demoMaskDatas.size() > 0) ) {
+	if (recordTriggered || (demoModeGenPreviews && demoMaskDatas.size() > 0)) {
 		demoModeRender(vidTex, mask_tex, mask_data);
 	}
 
 	// restore rendering state
 	gs_blend_state_pop();
+	if (logMode){
+		auto processEnd = NEW_TIMESTAMP;
+		auto elapsedMs = std::chrono::duration_cast<std::chrono::milliseconds>(processEnd - timestamp);
+		if (!sameFrameResults && !processedFrameResults.isSkipped()) {
+			lastActualTimestamp = timestamp;
+		}
 
+		int actualLatency = std::chrono::duration_cast<std::chrono::milliseconds>(processEnd - lastActualTimestamp).count();
+		int renderTime = std::chrono::duration_cast<std::chrono::microseconds>(processEnd - renderTimestamp).count();
+		int elapsedLatency = elapsedMs.count();
+		if (logOutput.is_open()) {
+			logOutput << "Latency: " << elapsedLatency << ", Latency # Frames: " << setprecision(1) << (float)elapsedLatency / 33.3f << ", " << processedFrameResults.to_string() << ", SameFrameResults: " << B2S(sameFrameResults) << endl; 
+			logOutput <<"Actual Latency: "<< setprecision(1) << actualLatency <<", Act. Latency  # Frames: %.1f" << (float)actualLatency << std::endl;
+			logOutput << "Render Time: " << renderTime << " micro secs." << std::endl;
+			logOutput.flush();
+		}
```

Comment:
Could you change the log format slightly so that the Measurement names only appear at the top of the log and they are all in a single line? And each set of measurements should be a new single line too so that it is easy to view in excel type applications.

Reply:
Fixed